test/api.test.ts

```diff
@@ -0,0 +1,106 @@
+import assert from 'assert';
+import supertest from 'supertest';
+import app from '../src/app';
+import mongoose from 'mongoose';
+import userModel from '../src/models/userModel';
+
+beforeEach((done) => {
+  mongoose.connect("mongodb://localhost:27017/TestDB",
+  { useNewUrlParser: true, useUnifiedTopology: true },
+  () => done());
+});
+
+afterEach((done) => {
+  mongoose.connection.db.dropDatabase(() => {
+    mongoose.connection.close(() => done())
+  });
+});
+
+const createDbEntry = async () => {
+  await userModel.create({
+    firstName: 'Agent',
+    lastName: 'Smith',
+    email: 'agentsmith@matrix.org'
+  }, {
+    firstName: 'Agent2',
+    lastName: 'Smith2',
+    email: 'agentsmith2@matrix.org'});
+};
+
+describe('GET /users', () => {
+  it('should get all database entries', async () => {
+    await createDbEntry();
+
+    const response = await supertest(app).get('/api/v1/users');
+
+    assert.ok(response);
```

Comment:
use as few assertions as possible and utilise full power of supertest:

await supertest(app).get('').status(200).expect('', '').expect(body, {});...

and when asserting something without supertest, create a habit of using assert.deepStrictEqual full objects instead of a few properties.